app/models/transaction.rb

```diff
@@ -3,6 +3,7 @@ class Transaction < ApplicationRecord
 
   belongs_to :account
   belongs_to :category, optional: true
+  belongs_to :merchant, optional: true
```

Comment:
We should probably update these lines to:

```
belongs_to :category, optional: true, dependent: :nullify
belongs_to :merchant, optional: true, dependent: :nullify
```

Currently, there is a database constraint for categories that nullifies the `category` field on all transactions when it is deleted introduced by [this migration](https://github.com/maybe-finance/maybe/blob/main/db/migrate/20240404112829_change_transaction_category_delete_behavior.rb).

I'm still not 100% sure whether we should be doing this at the model level, DB level, or _both_.  In the future, we may have other APIs modifying data (companion apps, etc.) which would lean towards handling it at the DB level.  But for now, I think we can just do it at the model level here.

Reply:
Doesn't the `has_many :transactions, dependent: :nullify` on the `Transaction::Category`/ `Transaction::Merchant` models already take care of this?

Reply:
Yep, you're 100% correct.  I was thinking about that backwards!